taxi/views.py

```diff
@@ -54,10 +92,29 @@ class ManufacturerDeleteView(LoginRequiredMixin, generic.DeleteView):
     success_url = reverse_lazy("taxi:manufacturer-list")
 
 
-class CarListView(LoginRequiredMixin, generic.ListView):
+class CarListView(LoginRequiredMixin, SearchMixin, generic.ListView):
     model = Car
     paginate_by = 5
     queryset = Car.objects.select_related("manufacturer")
+    search_placeholder = "Search by model"
+    model_search_field = "model"
+
+    # def get_context_data(self, *, object_list=None, **kwargs):
+    #     context = super(CarListView, self).get_context_data(**kwargs)
+    #     query = self.request.GET.get("query", "")
+    #     context["search_form"] = SearchForm(
+    #         initial={"query": query},
+    #         placeholder="Search by model"
+    #     )
+    #     return context
+    #
+    # def get_queryset(self):
+    #     queryset = self.queryset
+    #     form = SearchForm(self.request.GET, placeholder="Search by model")
+    #     if form.is_valid():
+    #         return queryset.filter(
+    #         model__icontains=form.cleaned_data["query"])
+    #     return queryset
```

Comment:
Why you commented this?

Reply:
Because I originally did the tutorial, but then I saw the duplication of code and it became interesting to try to make a generic search mixin that would be universal across different models. So I made it, but I left a comment for myself, so that if the mixin doesn't fit, I can quickly replace it.